upgrade/action.yaml

```diff
@@ -0,0 +1,126 @@
+name: Trunk Upgrade
+author: trunk.io
+description: Upgrade trunk and its tools
+
+branding:
+  icon: check
+  color: green
+
+inputs:
+  trunk-path:
+    description:
+      Path to Trunk Launcher. If not provided, we'll look for it the repo root, `.trunk/bin` and
+      `tools/`. If it can't be found anywhere and is not provided explicitly, we'll download it on
+      demand.
+    required: false
+
+  arguments:
+    description: Extra arguments to pass to trunk upgrade
+    required: false
+
+  base:
+    description: The base branch to create a PR against
+    required: false
+    default: main
+
+  github-token:
+    description:
+      A GitHub token to allow created PRs to run pull_request workflows. See
+      https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs
+      for more information.
+    required: false
+    default: ${{ github.token }}
+
+  reviewers:
+    description: A comma or newline separated list of GitHub reviewer usernames
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Locate trunk
+      shell: bash
+      run: |
+        # Locate trunk
+        ${GITHUB_ACTION_PATH}/../locate_trunk.sh
+      env:
+        INPUT_TRUNK_PATH: ${{ inputs.trunk-path }}
+
+    - name: Detect trunk
```

Comment:
replace this with https://github.com/trunk-io/trunk-action/blob/213102777d4a1c2687621d0035edb15664cbc493/action.yaml#L179-L190